File: sdkwork-backend-sdk-java/src/main/java/com/sdkwork/backend/api/AiMusicGenerationApi.java
package com.sdkwork.backend.api;

import com.sdkwork.backend.http.HttpClient;
import com.sdkwork.backend.model.*;
import java.util.List;
import java.util.Map;

public class AiMusicGenerationApi {
    private final HttpClient client;
    
    public AiMusicGenerationApi(HttpClient client) {
        this.client = client;
    }

    /** Create music generation task */
    public PlusApiResultGenerateMusicVO create(GenerateMusicForm body) throws Exception {
        return (PlusApiResultGenerateMusicVO) client.post(ApiPaths.backendPath("/generation/music/create"), body);
    }

    /** Get music generation result */
    public PlusApiResultGenerateMusicVO getResult(String taskId) throws Exception {
        return (PlusApiResultGenerateMusicVO) client.get(ApiPaths.backendPath("/generation/music/result/" + taskId + ""));
    }
}

File: sdkwork-backend-sdk-swift/Sources/API/AiMusicGenerationApi.swift
import Foundation

public class AiMusicGenerationApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Create music generation task
    public func create(body: GenerateMusicForm) async throws -> PlusApiResultGenerateMusicVO? {
        let response = try await client.post(ApiPaths.backendPath("/generation/music/create"), body: body)
        return response as? PlusApiResultGenerateMusicVO
    }

    /// Get music generation result
    public func getResult(taskId: String) async throws -> PlusApiResultGenerateMusicVO? {
        let response = try await client.get(ApiPaths.backendPath("/generation/music/result/\(taskId)"))
        return response as? PlusApiResultGenerateMusicVO
    }
}
